background_scripts/main.js

```diff
@@ -182,6 +182,67 @@ const mkRepeatCommand = (command) => (function (request) {
   }
 });
 
+function nextZoomLevel(currentZoom, steps) {
+  const chromeLevels = [ // Chrome's default zoom levels.
+    0.25,
+    0.33,
+    0.50,
+    0.75,
+    0.80,
+    0.90,
+    1.00,
+    1.10,
+    1.25,
+    1.50,
+    1.75,
+    2.00,
+    2.50,
+    3.00,
+    4.00,
+    5.00,
+  ];
+  const firefoxLevels = [ // Firefox's default zoom levels.
+    0.30,
+    0.50,
+    0.67,
+    0.80,
+    0.90,
+    1.00,
+    1.20,
+    1.33,
+    1.50,
+    1.70,
+    2.00,
+    2.40,
+    3.00,
+    4.00,
+    5.00,
+  ];
+
+  let zoomLevels = chromeLevels; // Chrome by default
+  if (BgUtils.isFirefox()) {
+    zoomLevels = firefoxLevels;
+  }
+
+  if (steps === 0) { // Nothing
+    return currentZoom;
+  } else if (steps > 0) { // In
+    // Chrome sometimes returns values with floating point errors.
+    currentZoom += 0.0000001; // This is needed to solve floating point bugs in Chrome.
+    const nextIndex = zoomLevels.findIndex((level) => level > currentZoom);
+    // Properly handle index not found (we want not found to be after array, not -1).
+    const floorIndex = nextIndex < 0 ? zoomLevels.length : nextIndex - 1;
```

Comment:
If we didn't find the value in our list, IE, it was bigger than all of them, we want to say the index is past the end, not -1. This helps handle several edge cases.